eth/stagedsync/stage_headers.go

```diff
@@ -423,7 +424,27 @@ func handleNewPayload(
 		log.Info(fmt.Sprintf("[%s] New payload missing parent", s.LogPrefix()))
 		cfg.hd.SetPoSDownloaderTip(headerHash)
 		schedulePoSDownload(requestId, header.ParentHash, headerNumber-1, s, cfg)
-		return &engineapi.PayloadStatus{Status: remote.EngineStatus_SYNCING}, nil
+		currentHeadNumber := rawdb.ReadCurrentBlockNumber(tx)
+		if currentHeadNumber != nil && math.GetAbsoluteDifference(*currentHeadNumber, headerNumber) < 32 {
+			// We try waiting until we finish downloading the PoS blocks if the distance from the head is enough,
+			// so that we will perform full validation.
+			for i := 0; cfg.hd.PosStatus() == headerdownload.Syncing && i < 10; i++ {
+				time.Sleep(10 * time.Millisecond)
+			}
+			// If we downloaded the headers in time then save them and if we saved the current header, we return VALID
+			if cfg.hd.PosStatus() == headerdownload.Synced {
+				verifyAndSaveDownloadedPoSHeaders(tx, cfg, headerInserter)
+				bad, lastValidHash := cfg.hd.IsBadHeaderPoS(header.ParentHash)
```

Comment:
We don't need this special logic here, but instead we should proceed with the rest of `handleNewPayload` flow (`BeaconRequestList.Remove(requestId)`, `verifyAndSaveNewPoSHeader`, etc.)

Reply:
I think we can do it with ACCEPTED, but definetely not with INVALID as said check is not used later on.